### ISSUE TITLE
Chroot uses older libraries than provided by the steam runtime.

### ISSUE THREAD

**Issue by jwyatt-feral** [ValveSoftware/steam-runtime#88]
The steam runtime provides libnss 3.26, however the the chroot only provides libnss 3.19. This prevents code that relies on this from being built due to undefined symbols, even though said code would run fine in the steam-runtime.

It's possible to work around this by allowing undefined symbols, however this introduces the risk that symbols not provided by the steam runtime will be used without a build error.

Some other libraries appear to have also been upgraded without upgrading the chroot (such as SDL2, libstdc++, libva,...) and a lot of packages seem to be a revision behind (+srt4 instead of +srt5 etc).

**Comment by TTimo** [comment]
Hello @jwyatt-feral, have you checked scout_beta ?

**Comment by jwyatt-feral** [comment]
@TTimo - it looks like scout_beta more closely resembles the current steam runtime, and does include this update. I assumed that scout_beta would correspond to the steam runtime beta, but judging by the libSDL2 package it's somewhere between the two? Do you happen to know if builds within the beta chroot are intended to run within the non-beta steam runtime?

(I think the current steam runtime is on SDL2.0.5, whereas the beta has SDL2.0.8, and scout_beta has SDL2.0.6)

**Comment by eddieray** [comment]
From what I see, the current Steam runtime includes SDL2.0.4, which is from Jan. 2016.  My game needs SDL2.0.6 or higher... so I have to include my own build of SDL (built against scout) to ensure the game has it with the normal Steam Client.

The scout_beta runtime seems not very useful if the Steam Client is not using that runtime... we have to build to target the normal Steam Client or our games will potentially break.  IMHO, in order to stay current with libraries that are in active development like SDL2, scout should include the latest official SDL2.x as it become available.

**Comment by Swyter** [comment]
Yeah, would be neat to have up-to-date _SDL2_ and _OpenAL Soft_ packages; many games depend on them. SDL2 2.0.5 is ancient and doesn't even support Vulkan or many other handy functions like `SDL_GameControllerGetVendor()` / `SDL_GameControllerGetProduct()`.

Version 2.0.6 is the minimum viable version right now:
https://discourse.libsdl.org/t/sdl-2-0-6-released/23109

Having to reimplement our own runtime on top of the Steam Runtime defeats a bit its purpose. We aren't talking about bleeding-edge functionality. These are basic components that are functionally obsolete.

**Comment by Plagman** [comment]
We have very recent builds of SDL in the shipping runtime, I think the repositories just haven't been enabled to match them yet; will look into it, but SDL 2.0.6 is safe to rely on. Not sure about openal-soft, is the version currently distributed by the Steam Client Beta too old for you?

**Comment by TTimo** [comment]
The beta client just updated with a new runtime, and the corresponding packages have been pushed to the scout and scout_beta apt sources. Please review if this addresses your problem.

Note that scout and scout_beta are currently the same and matching the beta client runtime. A handful of packages may be newer than the production client. Hopefully that won't cause major problems, the situation will normalize when the beta client is promoted in a few weeks.
